### DISCUSSION TITLE
COG Serial SPI 132x32 Graphic LCD Display No Backlight ST7567A

### DISCUSSION THREAD

**Discussion by markhen57** [olikraus/u8g2#1868] [Ideas]
BuyDisplay have this nice small (low cost!) LCD using the ST7567A with 132x32 glass. It works well with "U8G2_ST7567_PI_132X64_F_4W_SW_SPI u8g2(...", except half the display is not visible (and the RAM usage is 2x what is required).

Would it be possible to request this config be added to the release?

Thanks for great work

Mark

**Comment by olikraus** [comment]
> BuyDisplay

Any link?

> half the display is not visible

What does this mean? Any picture available (e.g. hello world with corresponding code)?

> Would it be possible to request this config be added to the release?

I need more information: Especially where is the visible area of your display located within the 132x65 memory of the controller.

I mean, if you use the "U8G2_ST7567_PI_132X64_F_4W_SW_SPI  constructor, how does the command look like, that draws a border in your display:
```
u8g2.drawFrame(0, y, 132, 32)
```
So the question is, which value is required for y so that the frame is fully visible for your display.

One more point: Can you also test "U8G2_ST7567_122X32_1_4W_SW_SPI"?
Of course some pixel will be missing on the right, but how will it look? Using U8G2_ST7567_122X32_1_4W_SW_SPI as a template for your display, would be a little bit easier.

**Comment by markhen57** [comment]
Thank you for fast reply! Sorry I did not provide all the info, I will try to answer all below:

https://www.buydisplay.com/cog-serial-spi-132x32-graphic-lcd-display-no-backlight-st7567a


By "half display is visible" I mean that since code vertical height is 64 but display height is 32, I can only see top 32 rows (or second 32 rows, depending on Rotation). The rows that are visible display correctly.

The following code gives the display in the attached picture ("no rotation"):

` u8g2.setFont(u8g2_font_logisoso32_tf);`
 ` u8g2.drawStr(0,32,"Line 1");`
 ` u8g2.drawStr(0,64,"Line 2");`
![no_rotation](https://user-images.githubusercontent.com/51063889/169913207-899d8833-e4ad-43e3-88e1-06ce5403889e.jpg)

Adding rotation by 180 with this code gives the attached picture ("with rotation"):

`  u8g2.setDisplayRotation(U8G2_R2);`
 ` u8g2.setFont(u8g2_font_logisoso32_tf);`  
 ` u8g2.drawStr(0,32,"Line 1"); `
  `u8g2.drawStr(0,64,"Line 2"); `
![with_rotation](https://user-images.githubusercontent.com/51063889/169913220-19f27afb-22f8-4237-bfaf-6c945cc98150.jpg)

Executing drawFrame with the following line (only; no rotation) gives the attached picture ("drawFrame" - the display is correctly outlined):

`u8g2.drawFrame(0, 0, 132, 32);`
![drawFrame](https://user-images.githubusercontent.com/51063889/169913440-64dce009-b24e-4546-9d0a-595471d88226.jpg)

I realize I can use the display and U8G2 fine by using the half of the display that is visible (depending on rotation), the main reason for my request is to reduce the size of the RAM buffer required!

I tried "U8G2_ST7567_122X32_1_4W_SW_SPI" but it does not display anything at all with this selection? I have tried several different text positions, fonts and frames? I have confirmed I'm using the entry with the exact same pin configurations that work for the 132x64 case:

`U8G2_ST7567_PI_132X64_F_4W_SW_SPI u8g2(U8G2_R0, /* clock=*/ 13, /* data=*/ 11, /* cs=*/ 7, /* dc=*/ 9, /* reset=*/ 8);`
`U8G2_ST7567_122X32_F_4W_SW_SPI u8g2(U8G2_R0, /* clock=*/ 13, /* data=*/ 11, /* cs=*/ 7, /* dc=*/ 9, /* reset=*/ 8); `

I will look at the code in "u8x8_d_st7567.c" to see if I can spot what might be different between these 2 cases.

Please let me know if there is any other information you require?

Many thanks for your help!

Mark

**Comment by olikraus** [comment]
I have created an issue (https://github.com/olikraus/u8g2/issues/1869)  from this topic. Unfortunately I am not able to work on this topic for the next view days.
